### ISSUE TITLE
Alert for Clashing StatsD Metrics that Cause The StatsD Exporter to Crash

### ISSUE THREAD

**Issue by omgitsbillryan** [department-of-veterans-affairs/va.gov-team#13265]
## The Problem
It's possible for StatsD metrics in the `vets-api` codebase to "clash", such as when the label names on the metric are inconsistent. When this happens, the `statsd-exporter` process that runs on the `vets-api` instances, crashes and restarts, which prevents new stats from being scraped by prometheus.

In summary, we lose reporting of many of the `vets-api` metrics.

## More Context
This occurred on Sept 3, 2020 - [link](https://dsva.slack.com/archives/C30LCU8S3/p1599164683086500)

[Metrics were introduced that clashed with existing metrics](https://github.com/department-of-veterans-affairs/vets-api/commit/f0e21fc33b8eef117c3265b190851e79d0c7a33d). This resulted in fatal error messages in the statsd-exporter log file:

```
time="2020-09-03T19:03:30Z" level=fatal msg="A change of configuration created inconsistent metrics for \"api_auth_saml_request\". You have to restart the statsd_exporter, and you should consider the effects on your monitoring setup. Error: a previously registered descriptor with the same fully-qualified name as Desc{fqName: \"api_auth_saml_request\", help: \"Metric autogenerated by statsd_exporter.\", constLabels: {type=\"mhv\",version=\"v0\"}, variableLabels: []} has different label names or a different help string" source="exporter.go:82" 
```
([Cloudwatch logs source](https://console.amazonaws-us-gov.com/cloudwatch/home?region=us-gov-west-1#logEventViewer:group=dsva-vagov-prod/var/log/prometheus/statsd_exporter.log;stream=deployment-vagov-prod-vets-api-server-20200903-190131-10.247.33.111;filter=restart;start=2020-09-02T19:54:38Z))

## Work to be Done
- [ ] Create a metric filter in Cloudwatch that looks for 
- [ ] Create an alert that notifies the ops/backend tools team when this occurs

**Comment by ericbuckley** [comment]
Another possibility worth research, can we configure `statsd-exporter` to not restart when an invalid set of tags are received?

**Comment by ericbuckley** [comment]
> Another possibility worth research, can we configure `statsd-exporter` to not restart when an invalid set of tags are received?

I believe we're still running [version 0.3.0 of statsd_exporter](https://github.com/department-of-veterans-affairs/devops/blob/6bb7a512265cec2996b2bd0107605bc76572526b/ansible/build/roles/prometheus/defaults/main.yml#L25), its possible that if we upgrade to a more recent version, [post 0.10.2](https://github.com/prometheus/statsd_exporter/blob/master/CHANGELOG.md#0102--2019-05-17), inconsistent labels won't cause the exporter to restart 🤷‍♂️ 

https://github.com/prometheus/statsd_exporter/pull/194

**Comment by omgitsbillryan** [comment]
Good idea. Even in the newest version, it still seems like having inconsistent labels is a bad thing, it just won't cause a restart. Probably best to upgrade and then create a metric filter on the new error message.
